src/main/java/de/tum/in/www1/artemis/service/compass/CompassService.java

```diff
@@ -221,68 +227,75 @@ public void addAssessment(long exerciseId, long submissionId, List<Feedback> mod
-                if (!isSupported(modelingExercise.getDiagramType())) {
-                    return;
-                }
-                // Round compass grades to avoid machine precision errors, make the grades more
-                // readable
-                // and give a slight advantage which makes 100% scores easier reachable
-                // see: https://confluencebruegge.in.tum.de/display/ArTEMiS/Feature+suggestions
-                // for more
-                // information
-                grade = roundGrades(grade);
-
-                // Save to database
-                List<Feedback> automaticFeedbackAssessments = engine.convertToFeedback(grade, modelId, result);
-                result.getFeedbacks().addAll(automaticFeedbackAssessments);
-                result.setHasFeedback(false);
-
-                result.setRatedIfNotExceeded(modelingExercise.getDueDate(), modelingSubmission.get().getSubmissionDate());
-                result.setAssessmentType(AssessmentType.AUTOMATIC);
-                double maxPoints = modelingExercise.getMaxScore();
-                // biased points
-                double points = Math.max(Math.min(grade.getPoints(), maxPoints), 0);
-                result.setScore((long) (points * 100 / maxPoints));
-                result.setCompletionDate(ZonedDateTime.now());
-                result.setResultString(points, modelingExercise.getMaxScore());
-
-                resultRepository.save(result);
-                engine.removeModelWaitingForAssessment(modelId, true);
-            }
-            else {
-                log.info("Model " + modelId + " got a confidence of " + grade.getConfidence() + " and a coverage of " + grade.getCoverage());
+        ModelingSubmission modelingSubmission = optionalModelingSubmission.get();
+
+        Result result = resultRepository.findDistinctWithFeedbackBySubmissionId(modelId)
+                .orElse(new Result().submission(modelingSubmission).participation(modelingSubmission.getParticipation()));
+
+        // only assess automatically when there is no manual assessment yet
+        if (result.getAssessmentType() != AssessmentType.MANUAL && result.getAssessor() == null) {
+            ModelingExercise modelingExercise = modelingExerciseRepository.findById(result.getParticipation().getExercise().getId())
+                    .orElseThrow(() -> new IllegalStateException("Exercise referenced in participation could not be found"));
+
+            // Workaround for ignoring automatic assessments of unsupported modeling exercise types TODO remove this after adapting compass
+            if (!isSupported(modelingExercise.getDiagramType())) {
+                return;
             }
+
+            // Round compass grades to avoid machine precision errors, make the grades more readable and give a slight advantage which makes 100% scores easier reachable
+            // see: https://confluencebruegge.in.tum.de/display/ArTEMiS/Feature+suggestions for more information
```

Comment:
this is a bad link because other Github users do not have access to this page

Reply:
I removed the link from the comment as there is JavaDoc for the `roundGrades` method describing the rounding mechanism.